api/proto/teleport/legacy/types/types.proto

```diff
@@ -991,6 +991,14 @@ message ProvisionTokenSpecV2 {
   ProvisionTokenSpecV2GitHub GitHub = 7 [(gogoproto.jsontag) = "github,omitempty"];
   // CircleCI allows the configuration of options specific to the "circleci" join method.
   ProvisionTokenSpecV2CircleCI CircleCI = 8 [(gogoproto.jsontag) = "circleci,omitempty"];
+  // AgentMatcherLabels is a set of labels to be used by agents to match on resources.
+  // When an agent uses this token, the agent should monitor resources that match those labels.
+  // For databases, this means adding the labels to `db_service.resources.labels`.
+  wrappers.LabelValues AgentMatcherLabels = 9 [
```

Comment:
Okay, apologies for not looking closely enough earlier, I think I see what's going on here now.

The problem I see here is that `AgentMatcherLabels` doesn't do what this comment says: if I create a token manually and set this, it will be completely ignored unless I use the join script.

My understanding: We are essentially using the `ProvisionToken` as a way to pass state from the web UI where the user puts the labels in, via the handler where the token and the join script URL is created, to the handler which generates the join script. I see this is already happening with `SuggestedLabels`, but it feels to me like an abuse of `ProvisionToken`. If this pattern continues we may end up duplicating much of `teleport.yaml` inside the `ProvisionToken`

I think it could make sense to put specific fields in the `ProvisionToken` if they were used regardless of whether the join script was used or not, e.g. if you set `SuggestedLabels` any node using that token should get those labels. Of course, that doesn't work because the node sets its own labels and it can't read the token itself, but that's beside the point.

There is another way to pass this state: the join script URL. We already use this to pass the token name and app names and URIs, for example. Was this considered?

Reply:
I do see the downside of long URLs and more complicated parsing if we were to pass via URL. I think there's a couple options:

1. Do what you have here, but rename and/or update doc comments to indicate that these are internal fields only that only have an effect on the join script
2. Pass via URL and have really long URLs
3. Create a new backend resource to pass state to the join script handler (probably overkill...)
4. Use a URL shortener (okay, not seriously)
5. Instead of adding more specific fields to `ProvisionToken`, create just one `JoinScriptParams` that holds e.g. a JSON or protobuf blob of everything you want to pass to the join script handler

Reply:
That's exactly it :pray: 

Well, my preference goes towards the 1st or 5th.

In regards to the 5th: I'm not too found of using the term `Script` because I think we should try to remove as much scripting as possible.

We could also reach a middle-ground between those two options: prepend the field name with `Suggested`
I think it's clear that we're not enforcing this property but it still exists.
Per my suggestion, this new field would become `SuggestedAgentMatcherLabels` (or something similar)

Reply:
I think `SuggestedAgentMatcherLabels` would be alright along with your updated doc comments on the fields

Reply:
✅